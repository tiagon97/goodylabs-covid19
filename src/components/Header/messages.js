const messages = {
  header: 'COVID-19 Info',
};

export default messages;
