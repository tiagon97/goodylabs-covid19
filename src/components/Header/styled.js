import styled from 'styled-components';

const StyledHeader = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: 80px;
  line-height: 80px;
  background-color: white;
  text-align: center;
  box-shadow: 0 10px 30px -10px hsla(0, 0%, 0%, 0.1);
  z-index: 998;
`;

export default StyledHeader;
