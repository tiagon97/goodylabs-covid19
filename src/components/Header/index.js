import React from 'react';

import Heading from '../Heading';
import messages from './messages';
import StyledHeader from './styled';

const Header = () => (
  <StyledHeader>
    <Heading main>{messages.header}</Heading>
  </StyledHeader>
);
export default Header;
