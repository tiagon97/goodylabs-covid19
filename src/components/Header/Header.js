import React from 'react';
import styled from 'styled-components';
import Heading from '../Heading/Heading';

const StyledHeader = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  height: 80px;
  line-height: 80px;
  background-color: white;
  text-align: center;
  box-shadow: 0 10px 30px -10px hsla(0, 0%, 0%, 0.1);
`;

const Header = () => (
  <StyledHeader>
    <Heading main>COVID-19 Info</Heading>
  </StyledHeader>
);
export default Header;
