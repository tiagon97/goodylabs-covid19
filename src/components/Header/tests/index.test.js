/**
 *
 * Tests for Header
 *
 */

import React from 'react';
import { render } from '@testing-library/react';

import Header from '../index';

describe('<Header />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Header />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Header />);
    expect(firstChild).toMatchSnapshot();
  });
});
