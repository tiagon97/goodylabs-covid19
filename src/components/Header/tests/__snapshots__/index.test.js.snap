// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should render and match the snapshot 1`] = `
<header
  class="sc-AxirZ eJvLfk"
>
  <h1
    class="sc-AxjAm iyJNFV"
  >
    COVID-19 Info
  </h1>
</header>
`;
